src/dependencies/FileDownloader.ts

```diff
@@ -21,33 +26,14 @@ export class FileDownloader implements DependencyInstaller {
 		const tempFile = fs.createWriteStream(temp.basePath);
 
 		try {
-			await new Promise((resolve, reject) => {
-				http.get(this.remoteUrl, (response) => {
-					if (response.statusCode !== 200) {
-						reject(`request to ${this.remoteUrl} failed with status code ${response.statusCode}`);
-					}
-
-					const totalSize = parseInt(response.headers["content-length"]!, 10);
-					let currentSize = 0;
-
-					progressListener(0, "Downloading…");
-					response.on("data", (chunk) => {
-						currentSize += chunk.length;
-						tempFile.write(chunk);
-						progressListener(currentSize / totalSize, "Downloading…");
-					});
-
-					response.on("end", () => {
-						tempFile.close();
-						resolve();
-					});
-
-					response.on("error", async (err) => {
-						reject(err);
-					});
-				});
-			});
-
+			progressListener(0, "Downloading…");
+			await pipeline(
+				got
+					.stream(this.remoteUrl)
+					.on('downloadProgress', (prog: GotProgress) => progressListener(prog.percent, "Downloading…")),
+				tempFile
+			);
+	
```

Comment:
Not an issue but I love how much simpler this has gotten thanks to `got`.

Reply:
Indeed! I first multiplexed over `http.get` and `https.get` but implementing some logic to following redirects was too annoying for me that's why I was looking out for libraries